projects/igniteui-angular/src/lib/services/transaction/utilities.ts

```diff
@@ -0,0 +1,104 @@
+export enum TransactionType {
+    ADD = 'add',
+    DELETE = 'delete',
+    UPDATE = 'update'
+}
+
+export interface Transaction {
+    id: any;
+    type: TransactionType;
+    newValue: any;
+}
+
+export interface State {
+    value: any;
+    recordRef: any;
+    type: TransactionType;
+}
+
+export interface TransactionService {
+    /**
+     * Adds provided  transaction with recordRef if any
+     * @param transaction Transaction to be added
+     * @param recordRef Reference to the value of the record in the data source related to the changed item
+     *
+     * @returns If operation is successful.
+     */
+    add(transaction: Transaction, recordRef?: any): boolean;
```

Comment:
And then there's no need for this method to return a bool anymore

Reply:
done